org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/configuration/internal/DefaultProjectConfigurationPersistence.java

```diff
@@ -53,9 +61,28 @@ private static ProjectConfiguration loadFromPreferencesApi(IProject project) {
     }
 
     private static ProjectConfiguration readProjectConfiguration(IProject project, PreferenceStore preferences) {
-        String projectDir = preferences.read(PREF_KEY_CONNECTION_PROJECT_DIR);
-        String gradleDistribution = preferences.read(PREF_KEY_CONNECTION_GRADLE_DISTRIBUTION);
-        return ProjectConfigurationProperties.from(projectDir, gradleDistribution).toProjectConfiguration(project);
+        String projectDir = preferences.readString(PREF_KEY_CONNECTION_PROJECT_DIR);
+        String distribution = preferences.readString(PREF_KEY_CONNECTION_GRADLE_DISTRIBUTION);
+        boolean overrideWorkspaceSettings = preferences.readBoolean(PREF_KEY_OVERRIDE_WORKSPACE_SETTINGS, false);
+        boolean buildScansEnabled = preferences.readBoolean(PREF_KEY_BUILD_SCANS_ENABLED, false);
+        boolean offlineMode = preferences.readBoolean(PREF_KEY_OFFLINE_MODE, false);
+
+        File rootDir = rootProjectFile(project, projectDir);
+        GradleDistribution gradleDistribution = GradleDistributionSerializer.INSTANCE.deserializeFromString(distribution);
+        if (overrideWorkspaceSettings) {
+            return ProjectConfiguration.from(project, rootDir, gradleDistribution, overrideWorkspaceSettings, buildScansEnabled, offlineMode);
```

Comment:
I don't think we need two factory methods here, we just always pass the values and correctly respect "overrideWorkspaceSettings" when saving again.